frontend/src/components/sampleTypes/SampeTypeDefinitionForm.jsx

```diff
@@ -0,0 +1,327 @@
+import FieldLabels from './fields/FieldLabels';
+import samplesAPI from '../../helpers/api/samples';
+import LoadingBackdrop from '../shared/LoadingBackdrop';
+import { StandardButton, LinkButton } from '../shared/Buttons';
+import utils from '../../helpers/utils';
+import AlertToast from '../shared/AlertToast';
+
+const useStyles = makeStyles((theme) => ({
+  root: {},
+  container: {
+    minWidth: 'lg',
+    overflow: 'auto',
+  },
+  title: {
+    fontSize: '2.5rem',
+    fontWeight: '700',
+    marginTop: theme.spacing(1),
+    marginBottom: theme.spacing(0.25),
+  },
+  inputName: {
+    fontSize: '1rem',
+    fontWeight: '700',
+  },
+  spaceBelow: {
+    marginBottom: theme.spacing(1),
+  },
+}));
+
+const initialSampleType = {
+  id: null,
+  name: '',
+  description: '',
+};
+
+const newFieldType = {
+  id: null,
+  name: '',
+  ftype: '',
+  required: false,
+  array: false,
+  choices: '',
+  allowable_field_types: [],
+};
+
+const SampleTypeDefinitionForm = ({ match }) => {
+  const classes = useStyles();
+  const [sampleTypeName, setSampleTypeName] = useState(initialSampleType.name);
+  const [sampleTypeDescription, setSampleTypeDescription] = useState(initialSampleType.description);
+  const [fieldTypes, setFieldTypes] = useState([newFieldType]);
+  const [sampleTypes, setSampleTypes] = useState([]);
+  const [objectTypes, setObjectTypes] = useState([]);
+  const [inventory, setInventory] = useState(0);
+  const [id, setId] = useState(null);
+  const [isLoading, setIsLoading] = useState(true);
+  const [disableSubmit, setDisableSubmit] = useState(false);
+  const [alertProps, setAlertProps] = useState({});
+
+  const ref = useRef();
+
+  /*  Get sample types top populate sample options menu
+      We cannot use async directly in useEffect so we create an async function that we will call
+      from w/in useEffect.
+      Our async function gets and sets the sampleTypes.
+      We only want to fetch data when the component is mounted so we pass an empty array as the
+      second argument to useEffect  */
+  useEffect(() => {
+    const fetchData = async () => {
+      const data = await samplesAPI.getTypes();
+      setSampleTypes(data.sample_types);
+      setIsLoading(false);
+    };
+
+    fetchData();
+  }, []);
+
+  useEffect(() => {
+    const fetchDataById = async () => {
+      const data = await samplesAPI.getTypeById(match.params.id);
+      setFieldTypes(data.field_types);
+      setSampleTypeDescription(data.description);
+      setSampleTypeName(data.name);
+      setObjectTypes(data.object_types);
+      setInventory(data.inventory);
+      setId(data.id);
+      setIsLoading(false);
+    };
+
+    match.params.id ? fetchDataById() : '';
+  }, []);
+
+  /*  Update allowSubmit state if name and Description change
+      Disable submit if name or description are empty */
+  useEffect(() => {
+    setDisableSubmit(!(!!sampleTypeName || !!sampleTypeDescription));
```

Comment:
this logic should be 
`setDisableSubmit(!sampleTypeName.trim() || !sampleTypeDescription.trim());`

Note that on the backend both name and description are required (and whitespace is truncated), so if either of the two fields are "empty" then the system will return a form error.  Catching it on the front-end is a nice addition to the UX, and the logic above is more accurate.